<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>introduction-to-SNPutils • SNPutils</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="introduction-to-SNPutils">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SNPutils</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/introduction-to-SNPutils.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/LabeoFD/SNPutils"><span class="fa fab fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">



<script src="introduction-to-SNPutils_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>introduction-to-SNPutils</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/LabeoFD/SNPutils/blob/HEAD/vignettes/introduction-to-SNPutils.Rmd" class="external-link"><code>vignettes/introduction-to-SNPutils.Rmd</code></a></small>
      <div class="d-none name"><code>introduction-to-SNPutils.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction-to-snputils">Introduction to SNPutils<a class="anchor" aria-label="anchor" href="#introduction-to-snputils"></a>
</h2>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>SNPutils is an R package designed to convert numerical genotype data (0, 1, 2) to allele pairs. The package provides tools for:</p>
<ul>
<li>Converting genotype codes to actual allele pairs</li>
<li>Validating allele assignments according to standard genetic rules</li>
<li>Generating standardized output formats (TYP and COM)</li>
<li>Handling indels (insertions/deletions) correctly</li>
<li>Performing quality control checks on genotype data</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install SNPutils from GitHub:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"LabeoFD/SNPutils"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using GitHub PAT from the git credential store.</span></span>
<span><span class="co">#&gt; Skipping install of 'SNPutils' from a github remote, the SHA1 (669c82dc) has not changed since last install.</span></span>
<span><span class="co">#&gt;   Use `force = TRUE` to force installation</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-concepts">Basic Concepts<a class="anchor" aria-label="anchor" href="#basic-concepts"></a>
</h2>
<p>In genetic data processing, genotypes are often represented as numerical codes:</p>
<ul>
<li>0: Homozygous for the first allele (A/A)</li>
<li>1: Heterozygous (A/B)</li>
<li>2: Homozygous for the second allele (B/B)</li>
</ul>
<p>These codes are convenient for computation. SNPutils converts these codes back to biologically meaningful allele pairs using annotation data that specifies which nucleotides (A, C, G, T) correspond to each allele.</p>
</div>
<div class="section level2">
<h2 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>SNPutils comes with example datasets to demonstrate its functionality:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LabeoFD/SNPutils" class="external-link">SNPutils</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_genotypes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_annotations</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Examine the genotype data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">example_genotypes</span><span class="op">)</span></span>
<span><span class="co">#&gt;   probeset_id Sample1 Sample2 Sample3</span></span>
<span><span class="co">#&gt; 1        SNP1       0       2       0</span></span>
<span><span class="co">#&gt; 2        SNP2       0       2       1</span></span>
<span><span class="co">#&gt; 3        SNP3       1       1       1</span></span>
<span><span class="co">#&gt; 4        SNP4       1       1       2</span></span>
<span><span class="co">#&gt; 5        SNP5       0       1       0</span></span>
<span><span class="co">#&gt; 6        SNP6       0       1       0</span></span></code></pre></div>
<p>The genotype data contains:</p>
<ul>
<li>A <code>probeset_id</code> column with SNP identifiers</li>
<li>Columns for each sample (Sample1, Sample2, Sample3) with genotype codes (0, 1, 2)</li>
</ul>
</div>
<div class="section level2">
<h2 id="examine-the-annotation-data">Examine the annotation data<a class="anchor" aria-label="anchor" href="#examine-the-annotation-data"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">example_annotations</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Probe.Set.ID Chromosome Physical.Position Allele.A Allele.B</span></span>
<span><span class="co">#&gt; 1         SNP1         10          43550342        A        C</span></span>
<span><span class="co">#&gt; 2         SNP2          X          73865503        A        C</span></span>
<span><span class="co">#&gt; 3         SNP3         22          99353247        T        C</span></span>
<span><span class="co">#&gt; 4         SNP4         17          52692942        A        C</span></span>
<span><span class="co">#&gt; 5         SNP5          2          66627319        A        C</span></span>
<span><span class="co">#&gt; 6         SNP6         15          36576624        A        C</span></span></code></pre></div>
<p>The annotation data contains:</p>
<ul>
<li>A <code>Probe.Set.ID</code> column matching the probeset_id values</li>
<li>Chromosome and position information</li>
<li>
<code>Allele.A</code> and <code>Allele.B</code> columns specifying the nucleotides</li>
</ul>
<p>The example data contains both standard SNPs and indels. For indels, the <code>Allele.A</code> column contains a deletion marker (“-”) and <code>Allele.B</code> contains the insertion sequence.</p>
</div>
<div class="section level2">
<h2 id="core-functionality">Core Functionality<a class="anchor" aria-label="anchor" href="#core-functionality"></a>
</h2>
</div>
<div class="section level2">
<h2 id="converting-genotypes-to-alleles">Converting Genotypes to Alleles<a class="anchor" aria-label="anchor" href="#converting-genotypes-to-alleles"></a>
</h2>
<p>The main function genotype2allele() converts numeric genotype codes to allele pairs:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert genotypes to alleles</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genotype2allele.html">genotype2allele</a></span><span class="op">(</span><span class="va">example_genotypes</span>, <span class="va">example_annotations</span><span class="op">)</span></span>
<span><span class="co">#&gt; No rule violations found - all rules are respected.</span></span>
<span></span>
<span><span class="co"># View the structure of the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ validation:List of 2</span></span>
<span><span class="co">#&gt;   ..$ allele_validation: tibble [20 × 12] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   .. ..$ probeset_id : chr [1:20] "SNP1" "SNP10" "SNP11" "SNP12" ...</span></span>
<span><span class="co">#&gt;   .. ..$ allele_a    : chr [1:20] "A" "T" "-" "-" ...</span></span>
<span><span class="co">#&gt;   .. ..$ allele_b    : chr [1:20] "C" "C" "A" "G" ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_indel    : logi [1:20] FALSE FALSE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_multibase: logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_AT_SNP   : logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_CG_SNP   : logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_other_SNP: logi [1:20] TRUE TRUE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule2_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule3_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule4_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule5_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "sorted")= chr "Probe.Set.ID"</span></span>
<span><span class="co">#&gt;   ..$ rule_violations  : tibble [0 × 13] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   .. ..$ probeset_id : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ allele_a    : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ allele_b    : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_indel    : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_multibase: logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_AT_SNP   : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_CG_SNP   : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_other_SNP: logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule2_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule3_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule4_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule5_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ violations  : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "sorted")= chr "Probe.Set.ID"</span></span>
<span><span class="co">#&gt;  $ results   :Classes 'data.table' and 'data.frame': 60 obs. of  3 variables:</span></span>
<span><span class="co">#&gt;   ..$ probeset_id    : chr [1:60] "SNP1" "SNP1" "SNP1" "SNP10" ...</span></span>
<span><span class="co">#&gt;   ..$ Sample_ID      : chr [1:60] "Sample1" "Sample2" "Sample3" "Sample1" ...</span></span>
<span><span class="co">#&gt;   ..$ allele_genotype: chr [1:60] "A/A" "C/C" "A/A" "C/C" ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, ".internal.selfref")=&lt;externalptr&gt;</span></span></code></pre></div>
<p>The <code><a href="../reference/genotype2allele.html">genotype2allele()</a></code> function returns a list with two components:</p>
<ul>
<li>
<code>validation</code>: Contains validation results and rule violations</li>
<li>
<code>results</code>: Contains the converted genotype data</li>
</ul>
<p>Let’s examine the conversion results:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Look at the first few rows of the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;    probeset_id Sample_ID allele_genotype</span></span>
<span><span class="co">#&gt;         &lt;char&gt;    &lt;char&gt;          &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:        SNP1   Sample1             A/A</span></span>
<span><span class="co">#&gt; 2:        SNP1   Sample2             C/C</span></span>
<span><span class="co">#&gt; 3:        SNP1   Sample3             A/A</span></span>
<span><span class="co">#&gt; 4:       SNP10   Sample1             C/C</span></span>
<span><span class="co">#&gt; 5:       SNP10   Sample2             T/T</span></span>
<span><span class="co">#&gt; 6:       SNP10   Sample3             C/C</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ validation:List of 2</span></span>
<span><span class="co">#&gt;   ..$ allele_validation: tibble [20 × 12] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   .. ..$ probeset_id : chr [1:20] "SNP1" "SNP10" "SNP11" "SNP12" ...</span></span>
<span><span class="co">#&gt;   .. ..$ allele_a    : chr [1:20] "A" "T" "-" "-" ...</span></span>
<span><span class="co">#&gt;   .. ..$ allele_b    : chr [1:20] "C" "C" "A" "G" ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_indel    : logi [1:20] FALSE FALSE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_multibase: logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_AT_SNP   : logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_CG_SNP   : logi [1:20] FALSE FALSE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ is_other_SNP: logi [1:20] TRUE TRUE FALSE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule2_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule3_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule4_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..$ rule5_check : logi [1:20] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "sorted")= chr "Probe.Set.ID"</span></span>
<span><span class="co">#&gt;   ..$ rule_violations  : tibble [0 × 13] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   .. ..$ probeset_id : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ allele_a    : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ allele_b    : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_indel    : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_multibase: logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_AT_SNP   : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_CG_SNP   : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ is_other_SNP: logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule2_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule3_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule4_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ rule5_check : logi(0) </span></span>
<span><span class="co">#&gt;   .. ..$ violations  : chr(0) </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, ".internal.selfref")=&lt;externalptr&gt; </span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "sorted")= chr "Probe.Set.ID"</span></span>
<span><span class="co">#&gt;  $ results   :Classes 'data.table' and 'data.frame': 60 obs. of  3 variables:</span></span>
<span><span class="co">#&gt;   ..$ probeset_id    : chr [1:60] "SNP1" "SNP1" "SNP1" "SNP10" ...</span></span>
<span><span class="co">#&gt;   ..$ Sample_ID      : chr [1:60] "Sample1" "Sample2" "Sample3" "Sample1" ...</span></span>
<span><span class="co">#&gt;   ..$ allele_genotype: chr [1:60] "A/A" "C/C" "A/A" "C/C" ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, ".internal.selfref")=&lt;externalptr&gt;</span></span></code></pre></div>
<p>For each genotype, the function has created an <code>allele_genotype</code> showing the actual nucleotide alleles.</p>
</div>
<div class="section level2">
<h2 id="validating-allele-assignments">Validating Allele Assignments<a class="anchor" aria-label="anchor" href="#validating-allele-assignments"></a>
</h2>
<p>SNPutils automatically validates allele assignments according to standard rules:</p>
<p><strong>Rule 1</strong>: SNPs are fixed on the forward strand of the design-time reference genome <strong>Rule 2</strong>: For AT/CG SNPs, alleles should be in alphabetical order <strong>Rule 3</strong>: For other SNPs, Allele.A should be A or T, and Allele.B should be C or G <strong>Rule 4</strong>: For indels, Allele.B should not be a deletion marker <strong>Rule 5</strong>: For multi-base alleles, alleles should be in alphabetical order</p>
<p>Let’s check if any validation issues were found:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check for validation issues</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"No validation issues found\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; No validation issues found</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="converting-to-typ-format">Converting to TYP Format<a class="anchor" aria-label="anchor" href="#converting-to-typ-format"></a>
</h2>
<p>The <code><a href="../reference/allele2typ.html">allele2typ()</a></code> function converts the results to TYP format: <strong>NOTE: It’s a TYP format missing the header (Pre-TYP)</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert to TYP format</span></span>
<span><span class="va">typ_format</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/allele2typ.html">allele2typ</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the TYP format data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">typ_format</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Sample_ID SNP_Name Allele1_Forward Allele2_Forward</span></span>
<span><span class="co">#&gt;       &lt;char&gt;   &lt;char&gt;          &lt;char&gt;          &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:   Sample1     SNP1               A               A</span></span>
<span><span class="co">#&gt; 2:   Sample2     SNP1               C               C</span></span>
<span><span class="co">#&gt; 3:   Sample3     SNP1               A               A</span></span>
<span><span class="co">#&gt; 4:   Sample1    SNP10               C               C</span></span>
<span><span class="co">#&gt; 5:   Sample2    SNP10               T               T</span></span>
<span><span class="co">#&gt; 6:   Sample3    SNP10               C               C</span></span></code></pre></div>
<p>The TYP format includes:</p>
<ul>
<li>
<code>Sample_ID</code>: Sample identifier</li>
<li>
<code>SNP_Name</code>: SNP identifier</li>
<li>
<code>Allele1_Forward</code> and <code>Allele2_Forward</code>: The two alleles</li>
</ul>
</div>
<div class="section level2">
<h2 id="handling-indels">Handling Indels<a class="anchor" aria-label="anchor" href="#handling-indels"></a>
</h2>
<p>Let’s focus on how SNPutils handles indels:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter for indels (SNP11-SNP20 in our example data)</span></span>
<span><span class="va">indel_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">results</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"SNP1[1-9]|SNP20"</span>, <span class="va">probeset_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">indel_results</span><span class="op">)</span></span>
<span><span class="co">#&gt;    probeset_id Sample_ID allele_genotype</span></span>
<span><span class="co">#&gt;         &lt;char&gt;    &lt;char&gt;          &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:       SNP11   Sample1             D/D</span></span>
<span><span class="co">#&gt; 2:       SNP11   Sample2             D/D</span></span>
<span><span class="co">#&gt; 3:       SNP11   Sample3             D/I</span></span>
<span><span class="co">#&gt; 4:       SNP12   Sample1             I/I</span></span>
<span><span class="co">#&gt; 5:       SNP12   Sample2             I/I</span></span>
<span><span class="co">#&gt; 6:       SNP12   Sample3             D/I</span></span></code></pre></div>
<p>For indels, the genotype codes are interpreted as:</p>
<ul>
<li>0: D/D (homozygous deletion)</li>
<li>1: D/I (heterozygous deletion/insertion)</li>
<li>2: I/I (homozygous insertion)</li>
</ul>
</div>
<div class="section level2">
<h2 id="working-with-your-own-data">Working with Your Own Data<a class="anchor" aria-label="anchor" href="#working-with-your-own-data"></a>
</h2>
<p>To use SNPutils with your own data, follow these steps:</p>
<ol style="list-style-type: decimal">
<li>Read Your Genotype Data Use <code><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">readr::read_delim()</a></code> or similar to read your genotype data, especially if you need to skip header lines:</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read genotype data</span></span>
<span><span class="va">genotype_data</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_delim</a></span><span class="op">(</span></span>
<span>  <span class="st">"path/to/genotype.txt"</span>,</span>
<span>  delim <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>  skip <span class="op">=</span> <span class="fl">583</span>,  <span class="co"># Adjust based on your file structure</span></span>
<span>  col_names <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Read annotation data</span></span>
<span><span class="va">annotation_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  <span class="st">"path/to/annotation.csv"</span>,</span>
<span>  header <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">","</span></span>
<span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Check and Prepare Your Data</li>
</ol>
<p>Ensure your data has the required column names:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For genotype data, ensure there's a column named "probeset_id"</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"probeset_id"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">genotype_data</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Rename the appropriate column</span></span>
<span>  <span class="va">genotype_data</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="va">genotype_data</span>, probeset_id <span class="op">=</span> <span class="va">your_id_column_name</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># For annotation data, ensure there are columns named "Probe.Set.ID", "Allele.A", and "Allele.B"</span></span>
<span><span class="va">required_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Probe.Set.ID"</span>, <span class="st">"Allele.A"</span>, <span class="st">"Allele.B"</span><span class="op">)</span></span>
<span><span class="va">missing_cols</span> <span class="op">&lt;-</span> <span class="va">required_cols</span><span class="op">[</span><span class="op">!</span><span class="va">required_cols</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">annotation_data</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">missing_cols</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Required columns missing from annotation data: "</span>,</span>
<span>       <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">missing_cols</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Process Your Data Convert genotypes to alleles:</li>
</ol>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert genotypes to alleles</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/genotype2allele.html">genotype2allele</a></span><span class="op">(</span><span class="va">genotype_data</span>, <span class="va">annotation_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check for validation issues</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span>, <span class="st">"validation issues"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">validation</span><span class="op">$</span><span class="va">rule_violations</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Convert to TYP format</span></span>
<span><span class="va">typ_format</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/allele2typ.html">allele2typ</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Save Your Results Write the results to a file:</li>
</ol>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Write TYP format to a file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">typ_format</span>, <span class="st">"output_typ.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hedia Tnani.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
